src/components/Categories.jsx

```diff
@@ -1,34 +1,48 @@
 import { useEffect, useState } from "react";
+import { Link } from "react-router-dom";
 import { getCategories } from "../api";
+import "../Category-list.css";
 
-const Categories = () => {
-  const [cats, setCats] = useState();
+const Categories = ({ currCategory, setCurrCategory }) => {
+  const [categoryList, setCategoryList] = useState([]);
   const [isLoading, setIsLoading] = useState(true);
 
   useEffect(() => {
     getCategories().then((categories) => {
-      setCats(categories);
+      setCategoryList(categories);
+      setIsLoading(false);
     });
-  });
+  }, []);
 
   return (
     <section>
-      <h2 id="categories-h2">Categories</h2>
+      <h2 id="categories-h2">Categories.</h2>
       {isLoading ? (
         <div>
           <p id="loading-text">Loading Categories...</p>
         </div>
       ) : (
-        <ul id="categories-ul">
-          {categories.map((category) => {
-            <article key={category.slug}>
-              <li key={category.slug}>
-                <p>{category.slug}</p>
-                <p>{category.description}</p>
-              </li>
-            </article>;
+        <section id="category-card-container">
+          {categoryList.map((category) => {
+            return (
+              <article id="category-list-card">
+                <Link
+                  to={`/${category.slug}`}
+                  key={category.slug}
+                  id={`category-link-${category.slug}`}
```

Comment:
Similarly here, the key needs to be provided for the parent article tag which is the main list element, not so much the link